### ISSUE TITLE
Only include production deps

### ISSUE THREAD

**Issue by lpil** [lpil/gleescript#1]
Currently all are included

**Comment by winterqt** [comment]
I’d like to take a stab at implementing this — I presume the best way is to parse both `gleam.toml` and `manifest.toml` to build up a set of all the prod deps required? (Unless there’s a library already out there to do this, shouldn’t be too big of a lift just to have in here though.)

**Comment by lpil** [comment]
Fab, thank you!

I don't know how you could do it from parsing those two files. Did you have an approach in mind? 🙏

**Comment by winterqt** [comment]
1. Parse `gleam.toml` to get the list of all prod deps
2. For each of them, recursively walk through `packages` to get the entire prod dep set

**Comment by lpil** [comment]
How would you handle dependencies that don't have a `gleam.toml`? Rebar and Mix deps, for example.

**Comment by winterqt** [comment]
You wouldn’t have to, I don’t think — you can get the entire dependency tree with just your Gleam project’s manifest.

**Comment by lpil** [comment]
It contains the names of the packages but not if they are dev deps or not.

edit: ah! You're saying that you'd check the parent to see how it has added the dep there! That makes sense!

**Comment by winterqt** [comment]
Exactly, yup -- we parse the `gleam.toml` to get the list of prod deps, then use `manifest.toml` to get the rest of the tree.

**Comment by lpil** [comment]
How would you tell if `dep_c` and `dep_d` are dev deps in this case?

```toml
[dependencies]
dep_a = "*"

[dev-dependencies]
dep_b = "*"
```

- root (Gleam)
  - dep_a (Not-Gleam)
    - dep_c (Not-Gleam)
  - dep_b (Not-Gleam)
    - dep_d (Not-Gleam)

**Comment by winterqt** [comment]
This would also happen for Gleam dependencies, unless we parse all of the `gleam.toml`s.

Maybe we should just add this info to `manifest.toml` to avoid the pain of trying to do all this in gleescript? Happy to do so -- can make an issue in the Gleam repo.

**Comment by winterqt** [comment]
(My understanding is that we could do this by parsing all the `gleam.toml`s if every dep is Gleam, but we'd then have to do the same for Rebar and Mix...)

**Comment by lpil** [comment]
Parsing rebar config is quite doable, but Mix config is Elixir code and can take any form at all. It could be evaluated though!

**Comment by winterqt** [comment]
Hex exposes whether a given dependency is a development one or not [0] — would we be better off using that within Gleam? Or should we just parse/evaluate the relevant configurations in gleescript?

[0] which also begs the question of if we include development dependencies in manifest.toml for transitive deps, I assume we do?
